include/alpaka/kernel/KernelBundle.hpp

```diff
@@ -0,0 +1,65 @@
+/* Copyright 2022 Benjamin Worpitz, Bert Wesarg, René Widera, Sergei Bastrakov, Bernhard Manfred Gruber, Mehmet
+ * Yusufoglu SPDX-License-Identifier: MPL-2.0
+ */
+
+#pragma once
+
+#include <alpaka/core/RemoveRestrict.hpp>
+
+#include <tuple>
+#include <type_traits>
+
+namespace alpaka
+{
+    //! \brief The class used to bind kernel function object and arguments together. Once an instance of this class is
+    //! created, arguments are not needed to be separately given to functions who need kernel function and arguments.
+    //! \tparam TKernelFn The kernel function object type.
+    //! \tparam TArgs Kernel function object invocation argument types as a parameter pack.
+    template<typename TKernelFn, typename... TArgs>
+    class KernelBundle
+    {
+    public:
+#if BOOST_COMP_CLANG
+#    pragma clang diagnostic push
+#    pragma clang diagnostic ignored "-Wdocumentation" // clang does not support the syntax for variadic template
+                                                       // arguments "args,...". Ignore the error.
+#endif
+        //! \param kernelFn The kernel function-object
+        //! \param args,... The kernel invocation arguments.
+#if BOOST_COMP_CLANG
+#    pragma clang diagnostic pop
+#endif
+        KernelBundle(TKernelFn const& kernelFn, TArgs&&... args)
+            : m_kernelFn(kernelFn)
+            , m_args(std::forward<TArgs>(args)...)
+        {
+        }
+
+        //! The function object type
+        using KernelFn = TKernelFn;
+        //! Tuple type to encapsulate kernel function argument types and argument values
+        using ArgTuple = std::tuple<remove_restrict_t<std::decay_t<TArgs>>...>;
+
+        KernelFn m_kernelFn;
+        ArgTuple m_args;
```

Comment:
I wanted to give user some flexibility (since bundle creates some "states" which is not good for dynamic scenarios; ) May be some variables might need to be updated by the user.